examples/plot_calibration_curve.py

```diff
@@ -0,0 +1,134 @@
+"""
+==============================
+Probability Calibration curves
+==============================
+
+When performing classification one often wants to predict not only the class
+label, but also the associated probability. This probability gives some
+kind of confidence on the prediction. This example demonstrates how to display
+how well calibrated the predicted probabilities are and how to calibrate an
+uncalibrated classifier.
+
+The experiment is performed on an artificial dataset for binary classification
+with 100.000 samples (1.000 of them are used for model fitting) with 20
+features. Of the 20 features, only 2 are informative and 10 are redundant. The
+first figure shows the estimated probabilities obtained with logistic
+regression, Gaussian naive Bayes, and Gaussian naive Bayes with both isotonic
+calibration and sigmoid calibration. The calibration performance is evaluated
+with Brier score, reported in the legend (the smaller the better). One can
+observe here that logistic regression is well calibrated while raw Gaussian
+naive Bayes performs very badly. This is because of the redundant features
+which violate the assumption of feature-independence and result in an overly
+confident classifier, which is indicated by the typical transposed-sigmoid
+curve.
+
+Calibration of the probabilities of Gaussian naive Bayes with isotonic
+regression can fix this issue as can be seen from the nearly diagonal
+calibration curve. Sigmoid calibration also improves the brier score slightly,
+albeit not as strongly as the non-parametric isotonic regression. This can be
+attributed to the fact that we have plenty of calibration data such that the
+greater flexibility of the non-parametric model can be exploited.
```

Comment:
Changing the ordering of this paragraph with the one on the calibration of the SVM model might help makes things clearer: AFAIK Platt's sigmoid scaling was devised for margin-based models in the first place for which the normalized decision function exhibit the expected (non-inverted) sigmoid shape.